algviz/parser/structures.py

```diff
@@ -213,11 +214,33 @@ class TreeNode(DataStructure):
     """A node with some number of children in a fixed order.  Edges are implicit."""
     # A common superclass could be used for linked-list nodes, since linked
     # lists are just skinny trees
+
     def __init__(self, data, children=None, **kwargs):
         super().__init__(**kwargs)
         self.data = data
         self.children = [] if children is None else children
 
+    def is_leaf(self):
```

Comment:
FYI, you should expect to get `Null`-valued children, especially as placeholders in rigid n-ary trees.  E.g., in a 3-ary tree, a node with a middle child but no left or right children will have children `[Null, TreeNode(...), Null]`